### ISSUE TITLE
Random seeds value

### ISSUE THREAD

**Issue by hamanhbui** [DeLightCMU/RSC#13]
Dear authors, thank you for your supporting code for DG, it could be a good baseline for us to expand a new idea.
However, I have a question about your reported results. I am seeing your code have set "torch.manual_seed(0) torch.cuda.manual_seed(0)", this means that you fix seeds for your CUDA, but your papers said that you ran on 5 different times (I understand here is with different seeds for CUDA) then get the average.
I am asking this question because if removing 2 of those lines, I can not reproduce your results, especially when tuning on validation set (only around 81.2% accuracy). Could you please explain this problem?

Looking forward to your answer, thanks.

**Comment by Justinhzy** [comment]
Hi,
Thanks for your question.
I fix the random seed and code wrt probabilities for PACS experiments for 5 runs.
Do you mean changing 5 different random seeds and get the average?

**Comment by hamanhbui** [comment]
Yes, It is exactly what I mean.

**Comment by SirRob1997** [comment]
@hamanhbui has been mentioned before, results seem to be non-reproducible see #12 #2 #5

**Comment by SirRob1997** [comment]
It is worth mentioning that to have consistent results you need to set all randomness and not just cuda:

```
np.random.seed(SEED)
random.seed(SEED) 
torch.manual_seed(SEED)
torch.cuda.manual_seed(SEED)
```

Also substitute l.79 + 79 [here](https://github.com/DeLightCMU/RSC/blob/master/Domain_Generalization/data/data_helper.py#L78) with:

```
loader = torch.utils.data.DataLoader(dataset, batch_size=args.batch_size, shuffle=True, num_workers=4, pin_memory=True, drop_last=True, worker_init_fn=random.seed(SEED))
val_loader = torch.utils.data.DataLoader(val_dataset, batch_size=args.batch_size, shuffle=False, num_workers=4, pin_memory=True, drop_last=False, worker_init_fn=random.seed(SEED))
```

Obviously, this introduces seed tuning and I'm generally not convinced that the method consistently provides performances similar to what is reported in the paper.

**Comment by hamanhbui** [comment]
Many thanks for your suggestion @SirRob1997

**Comment by Justinhzy** [comment]
Hi,
Sorry for the late reply. I've been a little busy lately.
Please see my updated notice. Generally speaking, most DG datasets are small, you may need to turn the parameter for your environment, including random seed. Though it is not ideal, you may get a higher baseline to start with.
